### ISSUE TITLE
All node operations should accept only host IDs rather than endpoints

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#12277]
This is part of #6403

In particular:
* bootstrap:
  - no external change is required.
* replace node:
  - add `replace_host_first_boot` option
  - deprecate and then stop supporting `replace_address_first_boot`
* decommission:
  - no external change is currently required
  - add ignore_nodes option that accepts a list of host_ids
* remove node:
  - deprecate, then remove support for endpoints for ignore_nodes option

Internally, we want to carry host_id in the node operation RPCs too.

**Comment by kostja** [comment]
@bhalevy could you please clarify what exactly needs to be done in scope of this issue? Perhaps it's best to split it into multiple so that we can assign each to a correct milestone.

**Comment by kbr-scylla** [comment]
Blocked by https://github.com/scylladb/scylladb/issues/19218
We can probably do https://github.com/scylladb/scylladb/issues/19218 in Q3, then this one in Q4. Maybe both in Q3 if we manage to do 2 releases in Q3

**Comment by kbr-scylla** [comment]
Note: I don't think this is part of https://github.com/scylladb/scylladb/issues/6403. https://github.com/scylladb/scylladb/issues/6403 is more about internals, if we get IP address through API, we can still translate it to host ID on the outermost layer and then pass host ID to internals. So it shouldn't block https://github.com/scylladb/scylladb/issues/6403

**Comment by kbr-scylla** [comment]
> add replace_host_first_boot option

We already have replace_node_first_boot.
(As you observed in https://github.com/scylladb/scylladb/issues/19482)

**Comment by kbr-scylla** [comment]
> Blocked by https://github.com/scylladb/scylladb/issues/19218

Fixed in 6.2.

**Comment by dani-tweig** [comment]
@kbr-scylla , what is the status of this one? Will it make it to 6.3 (this week), or should it be postponed to the coming milestone versions?

**Comment by kbr-scylla** [comment]
It's not even in the Q4 plan, but backlog:
![Image](https://github.com/user-attachments/assets/fea11d6f-adf5-46c0-bfba-1d62c56afb53)
so it won't make it this ~week~ year.
